### PR TITLE
Sparsify data  and show human readable time

### PR DESCRIPTION
**Pull request by pgdr**
Added a function `sparsify` that only picks out every _n_ minutes from the data we show which makes the chart more sparsely populated.  This function also caps data at 100 (for now), since spikes over 100 should be considered "over the top".

Now we also return `isotime` from the API, and a more human readable `time` field used in the table.  This translation could be moved to javascript if we don't want to increase the size of the data returned from server.  Feedback most welcome.

### CODE REVIEW

**Comment by pgdr** [comment]
If you wonder, appx 6500 rows per week are ignored now, while apx 4500 rows per week is returned.  So we actually cut down the load quite significantly.

File: friskby/views/quick.py [hunk 1]
```diff
@@ -5,17 +5,37 @@
...
+    if not lst:
+        return []
+    ret = []
+    prev = None
+    ret.append(lst[0])
+    prev = lst[0]['timestamp_data']
+    for e in lst:
+        delta = e['timestamp_data'] - prev
+        if delta.total_seconds() > minutes*60:
+            e['value'] = min(cutoff, e['value'])
```
Comment (joakim-hove) [inline friskby/views/quick.py]: Could we use `nan` instead? Then I guess we will get "holes" in the graph.

File: friskby/views/quick.py [hunk 2]
```diff
@@ -66,9 +87,13 @@ def get(self , request):
...
                 'pm25list': data25list, 
                 'pm10list': data10list,
-                'time': time}
+                'time': time_pp,
+                'isotime': time}
             device_rows.append(row)
 
+        algo_end = dt.now()
+        algo_delta = algo_end - algo_start
+        print('total time used: %.2f sec' % algo_delta.total_seconds())
```
Comment (joakim-hove) [inline friskby/views/quick.py]: Debug message - I guess this should be removed.

File: friskby/views/quick.py [hunk 3]
```diff
@@ -46,8 +62,8 @@ def get(self , request):
             except Sensor.DoesNotExist:
                 continue
 
-            data25query = [x for x in data_all if x["sensor_id"] == pm25sensor.id]
-            data10query = [x for x in data_all if x["sensor_id"] == pm10sensor.id]
+            data25query = sparsify([x for x in data_all if x["sensor_id"] == pm25sensor.id], minutes=30, cutoff=100)
```
Comment (joakim-hove) [inline friskby/views/quick.py]: I think common name for this operation is `downsample`?

**Comment by joakim-hove** [comment]
The `sparsify` function could easily be tested!

**Comment by mortalisk** [comment]
Looks good. I agree that debug statement can be removed. Testing would be good. I propose we have another workshop.

File: friskby/views/quick.py (hunk 1 above)
Comment (pgdr) [inline friskby/views/quick.py]: But I wanted `>100` to mean that the sensor is saturated and thus an actual point in the graph (however misleading) feels better than a hole in the graph.

File: friskby/views/quick.py (hunk 2 above)
Comment (pgdr) [inline friskby/views/quick.py]: Actually, I wanted to keep it if you don't mind.  I've been writing these lines of codes now every time I do something in this part just to ensure that I don't make changes that make Quick (much) slower.

File: friskby/views/quick.py (hunk 3 above)
Comment (pgdr) [inline friskby/views/quick.py]: Will change to `downsample`.

**Comment by pgdr** [comment]
I will change name to `downsample` and I will add a test.  If it's okay by you, I would like to keep the "`timeit`" logging.

File: friskby/views/quick.py (hunk 2 above)
Comment (joakim-hove) [inline friskby/views/quick.py]: OK - then we keep it for now; but I guess a better solution would be to look into Django Debug Toolbar.

**Comment by pgdr** [comment]
I have to admit I'm very uncertain how to do the testing here.

Is it possible to, say, `from views.quick import downsample` so that I can test on a function level?  Or do I have to do everything through web calls?

**Comment by joakim-hove** [comment]
Yes - do that import.

**Comment by pgdr** [comment]
Haha, thanks.  But I tried it, and
```python
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "~/friskby/friskby/friskby/tests/test_quick_view.py", line 8, in <module>
    from views.quick import downsample
ImportError: No module named views.quick
```

**Comment by joakim-hove** [comment]
Try a leading sensors.

13. feb. 2017 19:39 skrev "Pål Grønås Drange" <notifications@github.com>:

**Pull request merged**